lua/cl_specdm_hud.lua

```diff
@@ -78,77 +80,78 @@ local function PaintBar(x, y, w, h, colors, value)
+		ShadowedText(tostring(health), "HealthAmmo", bar_width, health_y, COLOR_WHITE, TEXT_ALIGN_RIGHT, TEXT_ALIGN_RIGHT)
+
+		if ttt_health_label:GetBool() then
+			local health_status = util.HealthToString(health)
+			draw.SimpleText(L[health_status], "TabLarge", x + margin*2, health_y + bar_height/2, COLOR_WHITE, TEXT_ALIGN_LEFT, TEXT_ALIGN_CENTER)
+		end
+
+		if ply:GetActiveWeapon().Primary then
+			local ammo_clip, ammo_max, ammo_inv = GetAmmo(LocalPlayer())
+			if ammo_clip != -1 then
+				local ammo_y = health_y + bar_height + margin
+				PaintBar(x+margin, ammo_y, bar_width, bar_height, ammo_colors, ammo_clip/ammo_max)
+
+				local text = string.format("%i + %02i", ammo_clip, ammo_inv)
+				ShadowedText(text, "HealthAmmo", bar_width, ammo_y, COLOR_WHITE, TEXT_ALIGN_RIGHT, TEXT_ALIGN_RIGHT)
 			end
-			local text = "Ghost DM"
-			local traitor_y = y - 30
-			ShadowedText(text, "TraitorState", x + margin + 73, traitor_y, COLOR_WHITE, TEXT_ALIGN_CENTER)
-			local is_haste = HasteMode() and round_state == ROUND_ACTIVE
-			local endtime = GetGlobalFloat("ttt_round_end", 0) - CurTime()
-			local text
-			local font = "TimeLeft"
-			local color = COLOR_WHITE
-			local rx = x + margin + 170
-			local ry = traitor_y + 3
-			if is_haste then
-				local hastetime = GetGlobalFloat("ttt_haste_end", 0) - CurTime()
-				if hastetime < 0 then
-					text = L.overtime
-					font = "Trebuchet18"
-					ry = ry + 5
-					rx = rx - 3
-				else
-					local t = hastetime
-					text = util.SimpleTime(math.max(0, t), "%02i:%02i")
-				end
+		end
+
+		local text = "Ghost DM"
+		local traitor_y = y - 30
+		ShadowedText(text, "TraitorState", x + margin + 73, traitor_y, COLOR_WHITE, TEXT_ALIGN_CENTER)
+
+		local is_haste = HasteMode() and GetRoundState() == ROUND_ACTIVE
+		local endtime = GetGlobalFloat("ttt_round_end", 0) - CurTime()
+		local text
+		local font = "TimeLeft"
+		local color = COLOR_WHITE
+		local rx = x + margin + 170
+		local ry = traitor_y + 3
+
+		if is_haste then
+			local hastetime = GetGlobalFloat("ttt_haste_end", 0) - CurTime()
+			if hastetime < 0 then
+				text = L.overtime
+				font = "Trebuchet18"
+				ry = ry + 5
+				rx = rx - 3
 			else
-				text = util.SimpleTime(math.max(0, endtime), "%02i:%02i")
-			end
-			ShadowedText(text, font, rx, ry, color)
-			if is_haste then
-				dr.SimpleText(L.hastemode, "TabLarge", x + margin + 165, traitor_y - 8)
+				local t = hastetime
+				text = util.SimpleTime(math.max(0, t), "%02i:%02i")
 			end
-			return
+		else
+			text = util.SimpleTime(math.max(0, endtime), "%02i:%02i")
+		end
+
+		ShadowedText(text, font, rx, ry, color)
+
+		if is_haste then
+			dr.SimpleText(L.hastemode, "TabLarge", x + margin + 165, traitor_y - 8)
 		end
-		return old_DrawHUD(self)
 	end
-		
-end)
\ No newline at end of file
+end)
+
+hook.Add("HUDShouldDraw", "GhostHUD", function(n)
+	local p = LocalPlayer()
+	if name == "TTTSpecHUD" and p:IsPlayer() and p:IsGhost() then return false end		
```

Comment:
Is already fixed in his next commit: https://github.com/Tommy228/TTT_Spectator_Deathmatch/pull/44/commits/045fc17f2c1f7f8bc1ef40be48990af7a241a78d